_posts/2021-08-23-gsoc-2021-project.md

```diff
@@ -0,0 +1,208 @@
+
+The bug was fixed by adding an extra condition before it reaches the error. If the format is `None`, Dask now uses use a default `png` format.
+
+```python
+import dask
+import dask.array as da
+
+array = da.arange(10)
+dask.visualize(array, filename=None)  # success
+```
+
+Reference: [Pull request #7740 by Freyam Mehta *"Fixing calling .visualize() with filename=None"*](https://github.com/dask/dask/pull/7740)
+
+## HTML representations
+
+Dask makes use of HTML representations in several places, for example in Dask collections like the Array and Dataframe classes (for background reading, see [this blogpost](https://matthewrocklin.com/blog/2019/07/04/html-repr)).
+
+More recently, we've introduced HTML representations for high level graphs into Dask, and Jacob Tomlinson has implemented HTML representations in several places in the dask distributed library (for further reading, see [this other blogpost](https://blog.dask.org/2021/07/07/high-level-graphs#visualization)).
+
+During Freyam's Google Summer of Code project, he extended the HTML representations for Dask high level graphs to include images, and introduced two entirely new HTML representations to the dask distributed library.
+
+### Array images in HTML repr for high level graphs
+
+The HTML representation for dask high level graphs has been extended, and now includes SVG images of dask arrays at intermediate stages of computation.
+
+The motivation for this feature is similar to the motivation behind adding tooltips, discussed above. Users want easier ways to access information about the way a Dask computation changes as it moves through each stage of computation. We hope this improvement to the HTML representation for Dask high level graphs will provide an at a glance summary of array shape and chunk size at each stage.
+
+Example:
+
+```python
+import dask.array as da
+
+array = da.ones((10, 20), chunks=(5, 10))
+array = array.T
+
+array.dask  # shows the HTML representation in Jupyter
+```
+
+<img src="/images/gsoc21/7886.png" alt="Example: Array images now included in HTML representation of Dask high level graphs, pull request #7886" height=414 width=736>
+
+Reference: [Pull request #7886 by Freyam Mehta *"Add dask.array SVG to the HTML Repr"*](https://github.com/dask/dask/pull/7886)
+
+### New HTML repr for ProcessInterface class
+
+A new HTML representation has been created for the `ProcessInterface` class in [dask distributed](https://github.com/dask/distributed/). The `ProcessInterface` class is not intended to be used directly. Instead, this information should be accessed via SSH scheduler or workers.
+
+The HTML representation displays the status, address, and external address of the process.
+
+There are three possible status options:
+* Process created, not yet running (blue icon)
+* Process is running (green icon)
+* Process closed (orange icon)
+
+Example:
+
+```python
+from dask.distributed import LocalCluster, Client, SSHCluster
```

Comment:
@freyam Is this the code example that was used to generate the demo images? If not, please update the example code here so they match.